src/components/Form.jsx

```diff
@@ -269,10 +287,10 @@ export default class Form extends React.Component {
     /* Call custom onFocus handler */
     const onFocusHandler = fieldProps.get('onFocus');
     if (onFocusHandler) {
-      onFocusHandler({
+      this.handleCallback(onFocusHandler, {
```

Comment:
So then we can use it like so in the Form:

```js
// components/Form.jsx
import { withImmutable } from '../utils';

...

if (onFocusHandler) {
  withImmutable(onFocusHandler, {
    event,
    ...
  }, this.context);
}
```

And somehow like this in utility methods:
```js
// utils/fieldUtils/validateAsync.js
import withImmutable from '../withImmutable';

export default function validateAsync(..., form) {
  const payload = withImmutable(rule, { ... }, form.context);
}
```
I think that all utils which dispatch callbacks have access to `form` reference, so we can grab the form's context by `form.context`.